### PR TITLE
Updated error condition for authType and imporved logging for basic auth

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# PR Submission checklist

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/139|

# Common PR Checklist:

- [x] Have you made sure that the code compiles?
- [x] Have you commented your code, particularly in hard-to-understand areas
- [x] Did you run tests in a real Kubernetes cluster?
- [x] Have you maintained backward compatibility

## Description of your changes:
Updated error condition for authType, It will set to default value 0 if any wrong value is given in values.yaml
Updated logs for masking basic auth creds

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -674,12 +679,12 @@ func (c *client) executeWithRetryAuthenticate(ctx context.Context, method, uri s
...
 func FetchValueIndexForKey(l string, match string, sep string) (int, int, int) {
 
+	var startIndex, endIndex = -1, -1
 	if strings.Contains(l, match) {
-		if i := strings.Index(l, match); i != -1 {
-			if j := strings.Index(l[i+len(match):], sep); j != -1 {
-				return i, j, len(match)
-			}
+		startIndex = strings.Index(l, match)
+		if sep != "" {
```
Comment (randeepsharma) [inline api/api.go:685]: could do  startIndex != -1  && sep != "" to avoidgoing inside the if to run the strings operation

File: api/api.go [hunk 2]
```diff
@@ -210,10 +210,15 @@ func New(
...
 	opts *ClientOptions) (Client, error) {
 
-	if hostname == "" || username == "" || password == "" || authType > 1 {
+	if hostname == "" || username == "" || password == "" {
 		return nil, errNewClient
 	}
 
+	if authType != 0 && authType != 1 {
+		log.Warn(ctx, "AuthType can be 0 or 1. Setting it to default value 0")
+		authType = 0
```
Comment (bpjain2004) [inline api/api.go:219]: don't hardcode; define a constant for defaultAuthType

File: api/api.go [hunk 3]
```diff
@@ -210,10 +210,15 @@ func New(
 	verboseLogging uint, authType uint8,
 	opts *ClientOptions) (Client, error) {
 
-	if hostname == "" || username == "" || password == "" || authType > 1 {
+	if hostname == "" || username == "" || password == "" {
 		return nil, errNewClient
 	}
 
+	if authType != 0 && authType != 1 {
```
Comment (bpjain2004) [inline api/api.go:217]: compare against consts authTypeBasic and authTypeSessionBased

**Comment by bpjain2004** [comment]
Update the README and examples in it with new env var GOISILON_AUTHTYPE

File: api/api.go (hunk 1 above)
Comment (Sakshi-dell) [reply api/api.go]: Updated

File: api/api.go (hunk 2 above)
Comment (Sakshi-dell) [reply api/api.go]: Used authTypeBasic instead of creating a new constant

File: api/api.go (hunk 3 above)
Comment (Sakshi-dell) [reply api/api.go]: Updated

**Review by bpjain2004 — APPROVED** [review]

**Review by randeepsharma — APPROVED** [review]

**Pull request merged**